LeakCanary RuntimeException when there is no label set in AndroidManifest

### Description

When there is no label attribute set in AndroidManifest <application> tag, LeakCanary crashes on application startup.

### Steps to Reproduce

Any project without label attribute in <application> AndroidManifest

1. Make sure you don't have label attribute set
2. First install of the app without the label attribute
3. start the app 
4. RuntimeException 

**Expected behavior:** Expect LeakCanary to not crash

### Version Information

* LeakCanary version: issue first happened on beta-1, issue not happening on alpha-3 version
* Android OS version: 8.0.0
* Gradle version: gradle-5.4.1

Confirmed by removing the tag from the sample app:

```
2019-09-10 16:33:20.760 23106-23106/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.leakcanary, PID: 23106
    java.lang.RuntimeException: Unable to get provider leakcanary.internal.AppWatcherInstaller: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
        at android.app.ActivityThread.installProvider(ActivityThread.java:5856)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:5445)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5384)
        at android.app.ActivityThread.-wrap2(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1545)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
        at leakcanary.internal.InternalLeakCanary.addDynamicShortcut(InternalLeakCanary.kt:169)
        at leakcanary.internal.InternalLeakCanary.invoke(InternalLeakCanary.kt:83)
        at leakcanary.internal.InternalLeakCanary.invoke(InternalLeakCanary.kt:30)
        at leakcanary.internal.InternalAppWatcher.install(InternalAppWatcher.kt:70)
        at leakcanary.internal.AppWatcherInstaller.onCreate(AppWatcherInstaller.kt:19)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:1751)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:1726)
        at android.app.ActivityThread.installProvider(ActivityThread.java:5853)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:5445) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5384) 
        at android.app.ActivityThread.-wrap2(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1545) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6119) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) 

```

--- a/leakcanary-android-core/src/main/java/leakcanary/internal/InternalLeakCanary.kt
+++ b/leakcanary-android-core/src/main/java/leakcanary/internal/InternalLeakCanary.kt
@@ -158,16 +158,10 @@ internal object InternalLeakCanary : (Application) -> Unit, OnObjectRetainedList
       longLabel = leakActivityLabel
       shortLabel = leakActivityLabel
     } else {
-
       val firstLauncherActivityLabel = if (firstMainActivity.labelRes != 0) {
         application.getString(firstMainActivity.labelRes)
       } else {
-        val applicationInfo = application.applicationInfo
-        if (applicationInfo.labelRes != 0) {
-          application.getString(applicationInfo.labelRes)
-        } else {
-          applicationInfo.nonLocalizedLabel.toString()
-        }
+        application.packageManager.getApplicationLabel(application.applicationInfo)
       }
       val fullLengthLabel = "$firstLauncherActivityLabel $leakActivityLabel"
       // short label should be under 10 and long label under 25

